Composer/packages/client/src/components/BotRuntimeController/OpenWebChatButton.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import formatMessage from 'format-message';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import { useRecoilValue } from 'recoil';
+
+import { BotStatus } from '../../constants';
+import { botStatusState, dispatcherState } from '../../recoilModel';
+import TelemetryClient from '../../telemetry/TelemetryClient';
+
+type OpenWebChatButtonProps = {
+  projectId: string;
+  isRoot: boolean;
+};
+
+export const OpenWebChatButton: React.FC<OpenWebChatButtonProps> = ({ projectId, isRoot }) => {
+  const { setWebChatPanelVisibility: toggleWebChatPanel } = useRecoilValue(dispatcherState);
+  const currentBotStatus = useRecoilValue(botStatusState(projectId));
+
+  if (!isRoot) {
+    return null;
+  }
+
+  const onOpenWebChatClick = () => {
+    toggleWebChatPanel(true);
+    TelemetryClient.track('WebChatPaneOpened');
+  };
+
+  return currentBotStatus === BotStatus.connected ? (
```

Comment:
ux nit: shouldn't it be disabled when bot is not connected instead of taking it off screen all together? @mewa1024 
(maybe with a tooltip message explaining why it's disabled)

Reply:
sorry, what piece of UI does this refer to?

Reply:
![](https://user-images.githubusercontent.com/13004779/109054593-e7ba9680-7692-11eb-950e-f3611cb54f28.png)

Reply:
@mewa1024 its this piece of Open Web chat

Reply:
I feel like this is going to cause more confusion if we just disable it. In R12 we do not support opening webchat for the skill. Having it and disabling it would mean like an operation can enable it which is falsy.